File: engine/engine-model/src/main/scala/tools/packet/monster/CatchMonster.scala
package tools.packet.monster

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class CatchMonster(private var _objectId: Int, private var _success: Byte) extends PacketInput {
  def objectId: Int = _objectId

  def success: Byte = _success

  override def opcode(): SendOpcode = SendOpcode.CATCH_MONSTER
}

File: engine/engine-model/src/main/java/tools/packet/monster/CatchMonster.java
package tools.packet.monster;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record CatchMonster(Integer objectId, Byte success) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CATCH_MONSTER;
   }
}